Rubberduck.Parsing/VBA/RubberduckParserState.cs

```diff
@@ -113,22 +113,34 @@ public void RefreshDeclarationFinder()
 
         private void AddEventHandlers()
         {
-            VBProjects.ProjectAdded += Sinks_ProjectAdded;
-            VBProjects.ProjectRemoved += Sinks_ProjectRemoved;
-            VBProjects.ProjectRenamed += Sinks_ProjectRenamed;
-            VBComponents.ComponentAdded += Sinks_ComponentAdded;
-            VBComponents.ComponentRemoved += Sinks_ComponentRemoved;
-            VBComponents.ComponentRenamed += Sinks_ComponentRenamed;           
+            _vbProjects.AttachEvents();
+            _vbProjects.ProjectAdded += Sinks_ProjectAdded;
+            _vbProjects.ProjectRemoved += Sinks_ProjectRemoved;
+            _vbProjects.ProjectRenamed += Sinks_ProjectRenamed;
+            foreach (var project in _vbProjects)
+            using (project)
+            {
+                project.VBComponents.AttachEvents();
+                project.VBComponents.ComponentAdded += Sinks_ComponentAdded;
+                project.VBComponents.ComponentRemoved += Sinks_ComponentRemoved;
+                project.VBComponents.ComponentRenamed += Sinks_ComponentRenamed;
+            }
         }
 
         private void RemoveEventHandlers()
         {
-            VBProjects.ProjectAdded -= Sinks_ProjectAdded;
-            VBProjects.ProjectRemoved -= Sinks_ProjectRemoved;
-            VBProjects.ProjectRenamed -= Sinks_ProjectRenamed;
-            VBComponents.ComponentAdded -= Sinks_ComponentAdded;
-            VBComponents.ComponentRemoved -= Sinks_ComponentRemoved;
-            VBComponents.ComponentRenamed -= Sinks_ComponentRenamed;
+            _vbProjects.ProjectAdded -= Sinks_ProjectAdded;
+            _vbProjects.ProjectRemoved -= Sinks_ProjectRemoved;
+            _vbProjects.ProjectRenamed -= Sinks_ProjectRenamed;
+            foreach (var project in _vbProjects)
+            using (project)
+            {
+                project.VBComponents.ComponentAdded -= Sinks_ComponentAdded;
+                project.VBComponents.ComponentRemoved -= Sinks_ComponentRemoved;
+                project.VBComponents.ComponentRenamed -= Sinks_ComponentRenamed;
+                project.VBComponents.DetachEvents();
```

Comment:
Please note that you are unsubscribing and detaching from on different components collection wrappers than you subscribed and attached to.